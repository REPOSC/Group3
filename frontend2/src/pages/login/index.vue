<template>
  <div class="login">
    <form class="form-control">
      <div class="card">
      <p>用户名:</p>
      <input type='text' class="input"/>
      <p>{{ username }}</p>
      </div>
      <div class="card">
      <p>密码:</p>
      <input type='password' class="input"/>
      <p>{{ password }}</p>
      </div>
      <button @click="tobookshelf">登录</button>
    </form>
  </div>
</template>

<script>

export default {
  data() {
    return {
      username: '10000017',
      password: 'bwCHikKhC5BjPit'
    }
  },
  methods: {
    test: function() {
      console.log(this.username)
      console.log(this.password)
    },
    handlelogin: function() {
      let Fly = require('../../../node_modules/flyio/dist/npm/wx.js')
      let fly = new Fly()
      let save = this
      fly.post('http://192.168.55.33:8082/auth_student', qs.stringify({
        id: save.username,
        password: save.password
      })).then(function(response) {
        if (response.data.status === 'error') {
          wx.showModal({
            content: '账号或密码错误，请重新登录'
          })
        } else {
          wx.redirectTo({
            url: '../level/main'
          })
        }
      })
    }
  }
}
</script>

<style>
a {
  padding: 10px 28px;
  margin-bottom: 30px，40px;
  background-color: #e7e7e7;
  color: black;
  border: none;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
}
.card {
  display:inline;
}
.input {
  display: block;
  padding: 0 12px;
  margin-bottom: 5px;
  border: 1px solid #ccc;
}
.form-control {
  display: block;
  padding: 150px 12px;
  margin-bottom: 5px;
}
.login {
  display: block;
  padding: 0px 20px;
  margin-bottom: 5px;
}
</style>
