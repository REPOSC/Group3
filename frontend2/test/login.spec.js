jest.mock('qs', () => ({
  stringify: jest.fn(() => Promise.resolve(mockData1))
}));
import { mount } from '@vue/test-utils'
import login from '../src/pages/login/index'
import qs from 'qs';

describe('test login interface', () => {
  let wrapper,vm;

  beforeEach(() => {
    wrapper = mount (login);
    vm = wrapper.vm;
    wrapper.setData(
      {
        username: ['12345'],
        password: ['11111']
      })
  });

  it('test for inputing account', () => {
    expect(vm.username).toEqual(['12345'])
  });

  it('test for password inputing', () => {
    expect(vm.password).toEqual(['11111'])
  });

  it('test login button', () => {
    expect(wrapper.contains('button')).toBe(true);
  });

  it('test the showing of form', () => {
    expect(wrapper.contains('form')).toBe(true);
  });

  it('test the showing of p', () => {
    expect(wrapper.contains('p')).toBe(true);
  });

  it('test the showing of div', () => {
    expect(wrapper.contains('div')).toBe(true);
  });

  it('test clicking handlelogin function', () => {
    const mockFn = jest.fn();
    wrapper.setMethods({
      handlelogin: mockFn
    });
    wrapper.find('button').trigger('click');
    expect(mockFn).toBeCalled();
    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it('test for qs.stringify using', () => {
    wrapper.find('button').trigger('click');
    expect(qs.stringify).toBeCalled();
  });
})
