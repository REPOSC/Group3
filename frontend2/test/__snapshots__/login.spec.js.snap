// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试登录界面 为login单元测试增加快照 1`] = `
<div
  class="login"
>
  <form
    class="form-control"
  >
    <div
      class="card"
    >
      <p>
        账号:
      </p>
       
      <input
        class="input"
        type="text"
      />
    </div>
     
    <div
      class="card"
    >
      <p>
        密码:
      </p>
       
      <input
        class="input"
        type="password"
      />
    </div>
     
    <button>
      登录
    </button>
  </form>
</div>
`;
